cmake/AddQCWorkflow.cmake

```diff
@@ -0,0 +1,35 @@
+# Copyright CERN and copyright holders of ALICE O2. This software is distributed
+# under the terms of the GNU General Public License v3 (GPL Version 3), copied
+# verbatim in the file "COPYING".
+#
+# See http://alice-o2.web.cern.ch/license for full licensing information.
+#
+# In applying this license CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization or
+# submit itself to any jurisdiction.
+
+include_guard()
+
+#
+# o2_add_qc_workflow(workflow-name configuration-file SOURCES ...) generates
+# a configuration json file via o2-qc --config <configuration-file> --dump
+# so that it can be easily registered and deployed e.g. in the train infrastructure.
+#
+# The installed JSON file will be named <workflow-name>.json and installed
+# under share/dpl directory
+#
+
+function(o2_add_qc_workflow workflowName configurationFile)
+
+  set(jsonDumpFile ${workflowName}.json)
+  set(qcExecutable o2-qc)
+
+  install(CODE "message(\"Generating a workflow dump with the command: ${qcExecutable} -b --config json:/${CMAKE_INSTALL_PREFIX}/${configurationFile} --dump-workflow --dump-workflow-file ${CMAKE_CURRENT_BINARY_DIR}/${jsonDumpFile}\")")
+  install(CODE "execute_process(
```

Comment:
I did not know one could specify some code for the installation step. Interesting. The only caveat with this is that it is not explicitely dependent on the qcExecutable. I don't think it hurts here, as a change in the executable would probably not change the workflow.json produced here. 

You could consider using `add_custom_command` (https://cmake.org/cmake/help/latest/command/add_custom_command.html) to output a file and make it depend on the executable. This being said, if it works like that I am not sure it is worth the trouble.

Reply:
I tried it, but it complained that the target is in another library (or directory, i don't remember), that would be probably the cleanest way otherwise.
Anyway, this still doesn't work, it links to wrong libraries, so we still get segfaults.

Reply:
Ah, this specific version cannot even find the executable, it does if I add a `bin/` prefix, but then it links to wrong libraries